src/SpectatorView.Unity/Assets/SpectatorView/Scripts/StateSynchronization/StateSynchronizationPerformanceMonitor.cs

```diff
@@ -7,17 +7,37 @@
 using System.IO;
 using System.Linq;
 using UnityEngine;
+using UnityEngine.Profiling;
 
 namespace Microsoft.MixedReality.SpectatorView
 {
     public class StateSynchronizationPerformanceMonitor : Singleton<StateSynchronizationPerformanceMonitor>
     {
-        private struct StopWatchKey
+        public class MemoryUsage
+        {
+            public long TotalAllocatedMemoryDelta;

```

Comment:
>TotalAllocatedMemoryDelta [](start = 24, length = 25)

I'm not sure the "Delta" is necessary in this name. If I understand correctly, this represents the total allocated memory by some specific component/scope that's being measured. Is that correct?

Reply:
Ideally yes, but if there are subsequent allocations that occur at the same time you can end up with values that don't truly reflect the measure you were going for. I'm not sure if the delta in the name helps reflect this.